src/middleware/addUserContext.ts

```diff
@@ -44,13 +43,41 @@ const addUserContext = (
 		return;
 	}
 
-	selectOrCreateUser(stringToKeycloakId(keycloakUserId))
-		.then((user) => {
-			(req as AuthenticatedRequest).user = user;
-			next();
+	Promise.all([
+		createOrUpdateUser(db, null, {
+			keycloakUserId: stringToKeycloakId(keycloakUserId),
+		}),
+		removeEphemeralUserGroupAssociationsByUserKeycloakUserId(
```

Comment:
I don't think we should remove here. The reason is other queries might be in flight and silently fail due to groups not being present. That raises the question, then, of when/where/how to remove stale groups. Hmm.

Reply:
Maybe an expiration would solve it. Then the remove operation can safely run in a blanket manner (i.e. remove all rows where notAfter is before now). This assumes we can guess at how many seconds would suffice for the in-flight requests to finish. Perhaps six or sixty would work well enough.

Reply:
OK I have an idea: NEVER REMOVE -- but instead add a `not_after` column which is auto-populated / updated on insert to give the row a "life" of 1 hour (or whatever we think is reasonable).

Reply:
That sounds good to me. Shrinking the life shrinks the window for a query to finish successfully although maybe it is the window to start the query. Extending the life extends the amount of time it takes to revoke access by removing somebody from an organization. Our JWTs are around a half hour right now, probably should be back at five minutes, so IDK, I guess I'd recommend something more like one to thirty minutes.

Oh, or how about this: expire it when the JWT expires (with some minimum, such as two minutes)?

Reply:
JWT expiration is a good idea -- we might need to think carefully about how ephemeral permissions + bulk upload work, since those jobs are run via a queue and in theory might not be started so quickly, but I think we're OK for now.